lib/charms/vault_k8s/v0/juju_facade.py

```diff
@@ -422,71 +452,108 @@ def get_remote_app_relation_data(
             MultipleRelationsFoundError
         """
         relation = (
-            self.get_relation_by_id(relation_name, relation_id)
-            if relation_id
-            else self.get_relation_by_name(relation_name)
+            (self.get_relation_by_id(name, id) if id else self.get_relation_by_name(name))
+            if not relation
+            else relation
         )
         if not relation:
-            raise NoSuchRelationError(f"Relation {relation_name}:{relation_id} not found")
+            raise NoSuchRelationError(f"Relation {name}:{id} not found")
         if not relation.app:
             logger.warning("No remote application yet")
-            raise NoRemoteAppError(
-                f"Relation {relation_name}:{relation_id} has no remote application yet"
-            )
+            raise NoRemoteAppError(f"Relation {name}:{id} has no remote application yet")
         return relation.data.get(relation.app, {})
 
     def get_unit_relation_data(
-        self, relation_name: str, relation_id: int | None
+        self,
+        name: str,
+        id: int | None = None,
+        relation: Relation | None = None,
     ) -> RelationDataContent | dict[str, str]:
         """Get relation data from the remote unit databag.
```

Comment:
The docstring does not seem quite right, this would not be the remote unit

Reply:
✅